openslides/motions/migrations/0003_motion_comments.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from ..config_variables import get_config_variables
+
+
+def change_motions_comments(apps, schema_editor):
+    """
+    Index the comments fields in the config. Changing from an array to a dict with the
+    ids as keys. CHange all motions from an array for comments to a dict with the comments
+    field id as key to link motion comments and comments fields.
+    """
+    # We get the model from the versioned app registry;
+    # if we directly import it, it will be the wrong version.
+    ConfigStore = apps.get_model('core', 'ConfigStore')
+    Motion = apps.get_model('motions', 'Motion')
+
+    try:
+        config_comments_fields = ConfigStore.objects.get(key='motions_comments').value
+    except ConfigStore.DoesNotExist:
+        config_comments_fields = get_motions_comments_default()
+
+    comments_fields = {}
+    index = 0
+    for field in config_comments_fields:
+        index += 1
+        comments_fields[index] = field
+
+    try:
+        db_value = ConfigStore.objects.get(key='motions_comments')
+    except ConfigStore.DoesNotExist:
+        db_value = ConfigStore(key='motions_comments')
+    db_value.value = comments_fields
+    db_value.save()
```

Comment:
I am not sure whether skip_autoupdate is necessary here.

Reply:
No: `TypeError: save() got an unexpected keyword argument 'skip_autoupdate'`

Reply:
Ok. So nothing to do ... :-)

Reply:
Hm. I do not understand why this happens and not for motion, see below. I also think that we do not want to trigger autoupdate during migration (this caused many errors in former times ...).

Reply:
Strange: The debugger stepped directly into djangos models/base.py. I've tried to override the save method in the ConfigStore class, but this doesn't work. So I added this to the ConfigStore:
```
def mysave(self, *args, **kwargs):
        return super().save(*args, **kwargs)
```
And calling `mysave` instead of `save`. Now I get `AttributeError: 'ConfigStore' object has no attribute 'mysave'`. It seems that during the migration (or because of getting the model with `apps.get_model`) the model is not loaded correctly. I do not have a clue why..

Reply:
```
(Pdb) db_value.__class__
<class '__fake__.ConfigStore'>
(Pdb) db_value.__class__.__bases__
(<class 'django.db.models.base.Model'>,)

(Pdb) motion.__class__
<class '__fake__.Motion'>
(Pdb) motion.__class__.__bases__
(<class 'openslides.utils.models.RESTModelMixin'>, <class 'django.db.models.base.Model'>)
```
....??

Reply:
Hm. Maybe models from the core app (because we do migrations in the core app) cannot be loaded directly but as a Django Fake Model. But models from other apps can, because their migrations are already done ...

Reply:
If you think this is the solution, please add a short comment in the code.

Reply:
Ok.